onmt/inputters/text_utils.py

```diff
@@ -16,15 +41,22 @@ def text_sort_key(ex):
 def clean_example(maybe_example):
     maybe_example['src'] = {"src": ' '.join(maybe_example['src'])}
     # Make features part of src like
-    # {'src': {'src': ..., 'feat1': ...., 'feat2': ....}}
-    if 'src_feats' in maybe_example:
-        for feat_name, feat_value in maybe_example['src_feats'].items():
-            maybe_example['src'][feat_name] = ' '.join(feat_value)
-        del maybe_example['src_feats']
+    # {'src': {'src': ..., 'feats': ....}
+    maybe_example['src']['feats'] = \
+        [' '.join(x) for x in maybe_example["src_feats"]]
     if maybe_example['tgt'] is not None:
         maybe_example['tgt'] = {'tgt': ' '.join(maybe_example['tgt'])}
+        maybe_example['tgt']['feats'] = \
+            [' '.join(x) for x in maybe_example["tgt_feats"]]
     if 'align' in maybe_example:
         maybe_example['align'] = ' '.join(maybe_example['align'])
+
+    # Delete these fields as they are no longer needed
+    del maybe_example['src_original']
+    del maybe_example['tgt_original']
```

Comment:
I was actually wondering if we may need to keep those for future sanity checks, especially in the server.

Reply:
What do you have in mind?